core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -33,7 +33,8 @@ oppia.factory('UrlService', ['$window', function($window) {
       var params = {};
       var parts = this.getCurrentHref().replace(
         /[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
-          params[decodeURIComponent(key)] = decodeURIComponent(value);
+          params[decodeURIComponent(key)] = decodeURIComponent(
+            value).split('#')[0];
```

Comment:
I think this is error-prone. Could you instead check for the hash (and ignore it) when parsing the URL?

Reply:
It should be possible to find a way to represent the internal structure of a URL using regex, or perhaps to use JS functions that extract its different parts. Better to do one of those than pure string manipulation.

Reply:
I changed regex to ignore hash also, how about now?

Reply:
It's still wrong. Try "?a=3&b=2#c?def=g" for example.

It looks like this is error-prone too. I suggest you use a JavaScript or Angular method to strip the hash explicitly.

Reply:
Ok, Done, changed it according to [this](https://stackoverflow.com/a/5298684). How about now?